File: assignments/assignment-2-solution/src-csharp/TeaShop/Domain/Inventory/StarRating.cs
namespace Assignment2Solution.Domain.Inventory;

/// <summary>
///     Represents a star rating for a tea item.
/// </summary>
public record StarRating
{
    /// <summary>
    ///     Initializes a new instance of the <see cref="StarRating" /> record.
    /// </summary>
    /// <param name="rating">The star rating value (1-5).</param>
    /// <exception cref="ArgumentOutOfRangeException">Thrown when rating is not between 1 and 5.</exception>
    public StarRating(int rating)
    {
        //invariant: rating must be between 1 and 5
        if (rating is < 1 or > 5)
            throw new ArgumentOutOfRangeException(nameof(rating), "Rating must be between 1 and 5");

        Rating = rating;
    }

    /// <summary>
    ///     Gets the star rating value.
    /// </summary>
    public int Rating { get; }

    /// <summary>
    ///     Returns a string representation of the star rating.
    /// </summary>
    /// <returns>A string with the rating value and a star character.</returns>
    public override string ToString()
    {
        return $"{Rating}{new string('*', Rating)}";
    }
}

File: assignments/assignment-2-solution/src-java/src/main/java/assignment2solution/domain/inventory/StarRating.java
package assignment2solution.domain.inventory;

/**
 * Represents a star rating for a tea item.
 */
public final class StarRating {
    private final int rating;

    /**
     * Initializes a new instance of the {@link StarRating} class.
     *
     * @param rating The star rating value (1-5).
     * @throws IllegalArgumentException when rating is not between 1 and 5.
     */
    public StarRating(int rating) {
        if (rating < 1 || rating > 5) {
            throw new IllegalArgumentException("Rating must be between 1 and 5");
        }
        this.rating = rating;
    }

    /**
     * Gets the star rating value.
     */
    public int getRating() {
        return rating;
    }

    /**
     * Returns a string representation of the star rating.
     *
     * @return A string with the rating value and a star character.
     */
    @Override
    public String toString() {
        return rating + "*".repeat(rating);
    }
}
